src/base.ts

```diff
@@ -47,6 +48,39 @@ export abstract class BitArray {
   public abstract getBit(index: number): boolean;
   public abstract setBit(index: number, value: boolean): void;
   public abstract toBitfield(): Uint8Array;
+
+  public or(bitArray: BitArray): BitArray {
+    for(let i = 0; i < this.bitLength; i++) {
+      this.setBit(i, this.getBit(i) || bitArray.getBit(i));
+    }
+    return this;
+  }
+
+  public and(bitArray: BitArray): BitArray {
+    for(let i = 0; i < this.bitLength; i++) {
+      this.setBit(i, this.getBit(i) && bitArray.getBit(i));
+    }
+    return this;
+  }
+
+  public overlaps(bitArray: BitArray): boolean {
+    assert(this.bitLength === bitArray.bitLength);
+
+    if(this.bitLength === 0) {
+      return false;
+    }
+
+    for(let i = 0; i < this.byteArray.length; i++) {
+      //invert byte from other array and xor against original, then xor against other array
+      // if all bits are false, result is 0 and it doesn't overlap
+      const result = (~this.byteArray[i] ^ bitArray.byteArray[i]) & bitArray.byteArray[i];
```

Comment:
I guess this is more efficient than `if (this.getBit(i) === other.getBit(i)) return true;`?
Maybe add a comment that we're traversing byte by byte rather than bit by bit.

Reply:
should be. Picked it up from other clients. You mean to add method comment because I have comment one line above?